### PR TITLE
 - collected commits for improving non-linear solver newton.

### PR DESCRIPTION
**Pull request by wibraun**

### CODE REVIEW

**Comment by lochel** [comment]
Please fix the indentation before merging this pull request.

File: SimulationRuntime/c/simulation/solver/newtonIteration.c [hunk 1]
```diff
@@ -0,0 +1,787 @@
...
+  data->delta_f = (double*) calloc(size,sizeof(double));
+  data->delta_x_vec = (double*) calloc(size,sizeof(double));
+
+  data->factorization = 0;
+  data->calculate_jacobian = 1;
+  data->numberOfIterations = 0;
+  data->numberOfFunctionEvaluations = 0;
+
+
+  assertStreamPrint(NULL, 0 != *voiddata, "allocationNewtonData() voiddata failed!");
```
Comment (lochel) [inline SimulationRuntime/c/simulation/solver/newtonIteration.c]: This assertion seems to be redundant.

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2015-07-15_14-28-38](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/378/console).

File: SimulationRuntime/c/simulation/solver/newtonIteration.h [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+#include "simulation_data.h"
+#include "nonlinearSolverNewton.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct DATA_NEWTON
+{
+  int initialized; /* 1 = initialized, else = 0*/
```
Comment (lochel) [inline SimulationRuntime/c/simulation/solver/newtonIteration.h]: This attribute seems not to be used properly. Should it get removed?

**Comment by lochel** [comment]
What are the improvements to the newton solver?

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2015-07-15_15-16-46](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/379/console).

**Pull request merged**